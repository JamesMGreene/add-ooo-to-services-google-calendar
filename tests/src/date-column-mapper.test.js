// Local modules
const dateColumnMapper = require('../../src/date-column-mapper')

const sampleRow = [
  '',
  '',
  'Timezones',
  '1/1/2019',
  '1/2/2019',
  '1/3/2019',
  '1/4/2019',
  '1/5/2019',
  '1/6/2019',
  '1/7/2019',
  '1/8/2019',
  '1/9/2019',
  '1/10/2019',
  '1/11/2019',
  '1/12/2019',
  '1/13/2019',
  '1/14/2019',
  '1/15/2019',
  '1/16/2019',
  '1/17/2019',
  '1/18/2019',
  '1/19/2019',
  '1/20/2019',
  '1/21/2019',
  '1/22/2019',
  '1/23/2019',
  '1/24/2019',
  '1/25/2019',
  '1/26/2019',
  '1/27/2019',
  '1/28/2019',
  '1/29/2019',
  '1/30/2019',
  '1/31/2019',
  // Handles leading zeroes
  '2/01/2019',
  '2/02/2019',
  '02/03/2019',
  '02/04/2019'
  // etc...
]

describe('date-column-mapper', () => {
  it('returns expected values', () => {
    expect(sampleRow.map(dateColumnMapper)).toMatchSnapshot()
  })
})
