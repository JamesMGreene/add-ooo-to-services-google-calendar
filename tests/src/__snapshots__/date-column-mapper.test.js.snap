// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-column-mapper returns expected values 1`] = `
Array [
  Object {
    "col": "A",
    "value": null,
  },
  Object {
    "col": "B",
    "value": null,
  },
  Object {
    "col": "C",
    "value": null,
  },
  Object {
    "col": "D",
    "value": 2019-01-01T00:00:00.000Z,
  },
  Object {
    "col": "E",
    "value": 2019-01-02T00:00:00.000Z,
  },
  Object {
    "col": "F",
    "value": 2019-01-03T00:00:00.000Z,
  },
  Object {
    "col": "G",
    "value": 2019-01-04T00:00:00.000Z,
  },
  Object {
    "col": "H",
    "value": 2019-01-05T00:00:00.000Z,
  },
  Object {
    "col": "I",
    "value": 2019-01-06T00:00:00.000Z,
  },
  Object {
    "col": "J",
    "value": 2019-01-07T00:00:00.000Z,
  },
  Object {
    "col": "K",
    "value": 2019-01-08T00:00:00.000Z,
  },
  Object {
    "col": "L",
    "value": 2019-01-09T00:00:00.000Z,
  },
  Object {
    "col": "M",
    "value": 2019-01-10T00:00:00.000Z,
  },
  Object {
    "col": "N",
    "value": 2019-01-11T00:00:00.000Z,
  },
  Object {
    "col": "O",
    "value": 2019-01-12T00:00:00.000Z,
  },
  Object {
    "col": "P",
    "value": 2019-01-13T00:00:00.000Z,
  },
  Object {
    "col": "Q",
    "value": 2019-01-14T00:00:00.000Z,
  },
  Object {
    "col": "R",
    "value": 2019-01-15T00:00:00.000Z,
  },
  Object {
    "col": "S",
    "value": 2019-01-16T00:00:00.000Z,
  },
  Object {
    "col": "T",
    "value": 2019-01-17T00:00:00.000Z,
  },
  Object {
    "col": "U",
    "value": 2019-01-18T00:00:00.000Z,
  },
  Object {
    "col": "V",
    "value": 2019-01-19T00:00:00.000Z,
  },
  Object {
    "col": "W",
    "value": 2019-01-20T00:00:00.000Z,
  },
  Object {
    "col": "X",
    "value": 2019-01-21T00:00:00.000Z,
  },
  Object {
    "col": "Y",
    "value": 2019-01-22T00:00:00.000Z,
  },
  Object {
    "col": "Z",
    "value": 2019-01-23T00:00:00.000Z,
  },
  Object {
    "col": "AA",
    "value": 2019-01-24T00:00:00.000Z,
  },
  Object {
    "col": "AB",
    "value": 2019-01-25T00:00:00.000Z,
  },
  Object {
    "col": "AC",
    "value": 2019-01-26T00:00:00.000Z,
  },
  Object {
    "col": "AD",
    "value": 2019-01-27T00:00:00.000Z,
  },
  Object {
    "col": "AE",
    "value": 2019-01-28T00:00:00.000Z,
  },
  Object {
    "col": "AF",
    "value": 2019-01-29T00:00:00.000Z,
  },
  Object {
    "col": "AG",
    "value": 2019-01-30T00:00:00.000Z,
  },
  Object {
    "col": "AH",
    "value": 2019-01-31T00:00:00.000Z,
  },
  Object {
    "col": "AI",
    "value": 2019-02-01T00:00:00.000Z,
  },
  Object {
    "col": "AJ",
    "value": 2019-02-02T00:00:00.000Z,
  },
  Object {
    "col": "AK",
    "value": 2019-02-03T00:00:00.000Z,
  },
  Object {
    "col": "AL",
    "value": 2019-02-04T00:00:00.000Z,
  },
]
`;
